inference-engine/tests/functional/plugin/cpu/single_layer_tests/non_max_suppression.cpp

```diff
@@ -0,0 +1,451 @@
+                const int32_t boxId   = selected_indices_data[i+2];
+                const float score = selected_scores_data[i+2];
+                if (batchId == -1 || classId == -1 || boxId == -1)
+                    break;
+
+                actualList.emplace_back(batchId, classId, boxId, coordList[batchId][boxId], score);
+            }
+            std::sort(actualList.begin(), actualList.end(), compareBox);
+        }
+
+        std::vector<Box> intersectionList;
+        std::vector<Box> differenceList;
+        {
+            std::list<Box> tempExpectedList(expectedList.size()), tempActualList(actualList.size());
+            std::copy(expectedList.begin(), expectedList.end(), tempExpectedList.begin());
+            std::copy(actualList.begin(), actualList.end(), tempActualList.begin());
+            auto sameBox = [](const Box& boxA, const Box& boxB) {
+                return (boxA.batchId == boxB.batchId) && (boxA.classId == boxB.classId) && (boxA.boxId == boxB.boxId);
+            };
+
+            for (auto itA = tempActualList.begin(); itA != tempActualList.end(); ++itA) {
+                bool found = false;
+                for (auto itB = tempExpectedList.begin(); itB != tempExpectedList.end(); ++itB) {
+                    if (sameBox(*itA, *itB)) {
+                        intersectionList.emplace_back(*itB);
+                        tempExpectedList.erase(itB);
+                        found = true;
+                        break;
+                    }
+                }
+
+                if (!found) {
+                    differenceList.emplace_back(*itA);
+                }
+            }
+            differenceList.insert(differenceList.end(), tempExpectedList.begin(), tempExpectedList.end());
+
+            for (auto& item : differenceList) {
+                if ((item.rect.x1 == item.rect.x2) || (item.rect.y1 == item.rect.y2))
+                    continue;
+
+                float maxIou = 0.f;
+                for (auto& refItem : intersectionList) {
+                    maxIou = std::max(maxIou, iouFunc(item, refItem));
+
+                    if (maxIou > 0.3f) break;
+                }
+
+                ASSERT_TRUE(maxIou > 0.3f) << "MaxIOU: " << maxIou
+                    << ", expectedList.size(): " << expectedList.size() << ", actualList.size(): " << actualList.size()
+                    << ", intersectionList.size(): " << intersectionList.size() << ", diffList.size(): " << differenceList.size()
+                    << ", batchId: " << item.batchId << ", classId: " << item.classId << ", boxId: " << item.boxId
+                    << ", score: " << item.score << ", coord: " << item.rect.x1 << ", " << item.rect.y1 << ", " << item.rect.x2 << ", " << item.rect.y2;
+            }
+        }
+    }
+    std::vector<TargetShapeParams> targetInDims;
+    size_t inferRequestNum = 0;
+    int32_t maxOutBoxesPerClass;
+};
+
+TEST_P(NmsLayerCPUTest, CompareWithRefs) {
+    run();
```

Comment:
Please add `SKIP_IF_CURRENT_TEST_IS_DISABLED()` and `CheckPluginRelatedResults(executableNetwork, "NMS")` calls, since we still do not check if the plugin implementation is called and not the ngraph reference.